suitcase/utils/tests/conftest.py

```diff
@@ -105,7 +105,7 @@ def generate_data(RE, detector_list, event_type):  # noqa
         list of 'event_types'.
     '''
 
-    def _generate_data_func(plan, ignore=None):
+    def _generate_data_func(plan, ignore=None, md={}):
```

Comment:
Save for certain special cases, it's good to avoid mutable defaults. Instead we adopt the widely-used pattern suggested, for example, [in this post](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).

Reply:
I know better but it's like my brain refuses to believe what python does with default arguments. I have fixed it. I notice the nested function `_example_data_func()` defines the default argument `ignore=[]`. Is a nested function definition a special case?

Also I have a gripe about the variable `ignore`. I think `skip_tests_with` or similar would convey the intention better, if I have managed to understand the intention.

Reply:
I would support that rename. I think the mutable default argument for `ignore` was just an oversight. Can @awalter-bnl confirm that I'm not forgetting any special reason for a mutable default there?

Reply:
@danielballan is remembering correctly, there is no particular reason for the mutable default or the choice of name. Renaming will mean changes to a few suitcases, which use this argument, but that isn’t a reason to not do it.

Reply:
I found four suitcase repositories that use the `ignore` argument to example_data(...):
- suitcase-csv
- suitcase-specfile
- suitcase-nxstxm (but conftest.py was copied wholesale)
- suitcase-xdi (obviously)
I would like to do that rename and fix those repositories, except for suitcase-nxstxm, which could just switch to using suitcase-utils and eliminate the duplicate code.